src/pjrt.cpp

```diff
@@ -96,7 +97,10 @@ Rcpp::XPtr<rpjrt::PJRTBuffer> create_buffer_from_r_data(
 
   // Copy data based on R type
   // copy implicitly casts the types
-  if constexpr (std::is_same_v<T, float> || std::is_same_v<T, double>) {
+  if (TYPEOF(data) == RAWSXP) {
+    std::copy(RAW(data), RAW(data) + len,
```

Comment:
Same as in the `to_host` implementation, we don't need an intermediry buffer in this case.

Reply:
done!